src/__swaps__/screens/Swap/components/SwapBottomPanel.tsx

```diff
@@ -36,6 +36,15 @@ export function SwapBottomPanel() {
   const disabled = useDerivedValue(() => confirmButtonProps.value.disabled);
   const opacity = useDerivedValue(() => confirmButtonProps.value.opacity);
 
+  const [type, setType] = useState<'tap' | 'hold'>('tap');
+  useAnimatedReaction(
+    () => confirmButtonProps.value.type,
+    (current = 'tap') => {
+      'worklet';
+      runOnJS(setType)(current);
+    }
+  );
+
```

Comment:
I'm not a huge fan of using state in this component. Is it possible to pass a sharedvalue that adapts to the reviewprogress into the SwapActionButton where we have local state there? That way it's a little more isolated.

Reply:
Instead of conditionally rendering the long press button, I'd make the `GestureHandlerHoldButton` support both regular taps as well as long presses, and then use a shared value to control whether taps or long presses are enabled based on the current `NavigationStep`.